src/easylink/devtools/implementation_creator.py

```diff
@@ -241,6 +251,22 @@ def _extract_implemented_step(script_path: Path) -> str:
             )
         return steps[0]
 
+    @staticmethod
+    def _extract_has_custom_recipe(script_path: Path) -> str:
+        """Extracts whether the user has already written the recipe for this implementation.
+
+        The expectation is that this flag is specified within the script
+        as a comment of the format:
+
+        .. code-block:: python
+            # HAS_CUSTOM_RECIPE: true
+        """
+        has_custom_recipe = _extract_metadata("HAS_CUSTOM_RECIPE", script_path)
+        if len(has_custom_recipe) == 0:
+            return False
+        else:
+            return str(has_custom_recipe[-1]).strip().lower() in ["true", "yes"]
```

Comment:
`has_custom_recipe` will always have length 0 or 1, right?
```suggestion
            return str(has_custom_recipe[0]).strip().lower() in ["true", "yes"]
```

Reply:
We don't enforce anywhere that the user doesn't have _multiple_ of the same magic comment, but I'm fine with making this change (it's kind of undefined behavior anyway)